tests/README.md

```diff
@@ -114,42 +144,48 @@ To do that run test(s) with `--pdb` flag:
 make tests args="[TESTFILE]::[TESTNAME] --pdb"
 ```
 
-Alternatively you can add `breakpoint()` before the failing assertion
-and manually invoke the single test.
+Alternatively you can add `breakpoint()` before the failing assertion and manually invoke the single test.
+
+### Container Shell
+
+You can also drop into a shell in the ` tests` container via `docker compose run --entrypoint=sh tests`, and from there run `pytest` manually.
+
+## Adding a Test
+
+Currently, all tests in the `tests` directory are functional tests for the chalk binary. Any unit tests should NOT be added here -- they should go into the nim source code per nim testing conventions.
+
+All python tests must follow `pytest` conventions to be picked up by the test runner:
+
+- New test files should be added to the `tests/` directory as `test_filename.py`. A new test file should be created to test new functionality, for example if support has been added for a new codec.
+- Individual tests should be added to a test file as a new function as `test_functionname`. Each individual test in a test file should test some aspect of that test file's functionality; if the test you want to add doesn't fit, consider if it would be more appropriate to create a new test file and add it there instead.
+- A new test case for an existing test can be added via the pytest `paramatrize` fixture, if appropriate.
+
+### Datafile Location
+
+All new test files should added to the `tests/` directory, and any test data should be added to the `tests/data` directory.
```

Comment:
```suggestion
All new test files should be added to the `tests/` directory, and any test data should be added to the `tests/data` directory.
```

Reply:
does vale catch grammar issues also? :scream_cat:

Reply:
some basics I think if you enable appropriate styles (they have defaults for microsoft, google and redhat - https://vale.sh/generator/). this was just me but Im not English expert so I have no idea if its actually correct. my logic if it sounds right, its prolly right but what "right" actually is 🤷